crowbar_framework/lib/crowbar/upgrade_status.rb

```diff
@@ -155,6 +166,24 @@ def upgrade_steps_6_7
       ]
     end
 
+    # Return true if user is allowed to execute given step
+    # In normal cases, that should be true only for next step in the sequence.
+    # But for some cases, we allow repeating of the step that has just passed.
+    def step_allowed?(step)
+      return true if step == current_step
+      if [
+        :upgrade_prechecks,
+        :admin_repo_checks,
+        :nodes_repo_checks
+      ].include? step
+        # Allow repeating one of these steps if it was the last one finished
+        # and no other one has been started yet.
+        i = upgrade_steps_6_7.index step
+        return upgrade_steps_6_7[i + 1] == current_step && !running?(current_step)
```

Comment:
This doesn't match the description in comment above. We should be checking if it's not started instead of not running (hint: passed = !running but passed != !started).